raiden/network/transport/matrix/transport.py

```diff
@@ -470,6 +471,10 @@ def on_success(greenlet):
         self.greenlets.append(greenlet)
         return greenlet
 
+    def _schedule(self, period: float, func: Callable, *args, **kwargs):
+        gevent.spawn_later(0, func, *args, **kwargs)
+        gevent.spawn_later(period, self._schedule, period=period, func=func, *args, **kwargs)
```

Comment:
Can you explain what this function does? So what I understand is that it spawns `self._schedule` after the given `period`. But this spawn_later is called in the very same function? Just want to make sure, that this is not a loop

Reply:
Good point.  It is effectively an endless loop. 

This needs to stop when the transport stop event is set (or rather the AddressManager stop event, see #4333)

Reply:
so it is intended to be an endless loop?

Reply:
I'll add a docstring, good point

Reply:
> so it is intended to be an endless loop?

Yes, but only while the transport is running.

Reply:
Okay so should I add to `transport.greenlets` or will it be cleared if the services main loop is shutdown?

Reply:
Added them to transports greenlets.